src/App.jsx

```diff
@@ -14,33 +16,92 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ABC = 'alphabetically';
+const SORT_FIELD_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPrepereGoods(goods, { sortField, reversed }) {
+  let prepareGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField === SORT_FIELD_ABC) {
+    prepareGoods.sort((good1, good2) => good1.localeCompare(good2));
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (sortField === SORT_FIELD_LENGTH) {
+    prepareGoods.sort((good1, good2) => good1.length - good2.length);
+  }
+
+  if (reversed) {
+    prepareGoods = prepareGoods.reverse();
+  }
+
+  return prepareGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortedField] = useState('');
```

Comment:
`setter` function and state new to be the same, we only add `set` part to function
```suggestion
  const [sortField, setSortField] = useState('');
```